controllers/PageController.php

```diff
@@ -457,4 +466,140 @@ public function actionToggleNumbering(int $id)
+            throw new HttpException(404);
+        }
+
+        $submittedRevision = new WikiPageRevision();
+        $submittedRevision->revision = time();
+        $submittedRevision->content = $form->revision->content;
+        $submittedRevision->isCurrentlyEditing = true;
+
+        $mergedRevision = $page->createRevision();
+        $changedContentSepeartor = '**conflicting changes from '. $dateTime->format('Y-m-d H:i:s').'**';
+        $mergedRevision->content = $page->latestRevision->content.'<br><br>'.$changedContentSepeartor.'<br>'.$submittedRevision->content;
+        $mergedRevision->save();
+
+        return $this->redirect(Url::toWiki($page));
+    }
+
+    /**
+     * @param int $id
+     * @return $this|Response
+     * @throws HttpException
+     */
+    public function actionCreateCopy(int $id) {
+        $userIdentity = Yii::$app->user->identity->username;
+        $dateTime = new DateTime();
+
+        $page = $this->getWikiPage($id);
+        if (!$page) {
+            throw new HttpException(404, 'Wiki page not found!');
+        }
+
+        $parentId = $page->parent_page_id;
+
+        $form = (new PageEditForm(['container' => $this->contentContainer]))->forPage($id);
+        if (!$form->load(Yii::$app->request->post())) {
+            throw new HttpException(404);
+        }
+
+        $childPage = new WikiPage();
+        $childPage->title = $page->title.' conflicting copy of '. $userIdentity.' from '. $dateTime->format('Y-m-d H:i:s');
+        $childPage->parent_page_id = $parentId;
+        $childPage->content->contentcontainer_id= $page->content->contentcontainer_id;
+
+        if (!$childPage->save()) {
+            throw new HttpException(500, 'Failed to create the child page!');
+        }
+
+        if (!$childPage->id) {
+            throw new HttpException('Child page ID is not available after saving.');
+        }
+
+        $revision = new WikiPageRevision();
+        $revision->content = $form->revision->content;
+        $revision->wiki_page_id = $childPage->id;
+        $revision->revision = 1;
+        $revision->user_id = Yii::$app->user->id;
+
+        if (!$revision->save()) {
+            throw new HttpException('Failed to add content to the child page.');
+        }
+
+        return $this->redirect(Url::toWiki($page));
+    }
+
+    /**
+     * @param int $id
+     * @return $this|Response
+     * @throws HttpException
+     */
+    public function actionEditingStatus(int $id)
+    {
+        $page = $this->getWikiPage($id);
+        if (!$page) {
+            throw new HttpException(404, 'Wiki page not found!');
+        }
+
+        $user = Yii::$app->user->identity->username;
+
+        return $this->asJson([
+            'success' => true,
+            'isEditing' => $page->isEditing(),
+            'body' => $page->is_currently_editing.' '.Yii::t('WikiModule.base', 'is already editing.<br> Editing it would cause conflict. Do you really want to continue?'),
```

Comment:
For the name in the message, we prefer the following format: <first name> <last name> (username).
E.g. Max Mustermann (max.mustermann)